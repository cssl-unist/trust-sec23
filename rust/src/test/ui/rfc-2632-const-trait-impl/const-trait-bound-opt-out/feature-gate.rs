// revisions: stock gated
// gate-test-const_trait_bound_opt_out

#![cfg_attr(gated, feature(const_trait_bound_opt_out))]
#![allow(incomplete_features)]
#![feature(rustc_attrs)]
#![feature(const_fn)]

trait T {
    const CONST: i32;
}

const fn get_assoc_const<S: ?const T>() -> i32 { <S as T>::CONST }
//[stock]~^ ERROR `?const` on trait bounds is experimental

#[rustc_error]
fn main() {} //[gated]~ ERROR fatal error triggered by #[rustc_error]
