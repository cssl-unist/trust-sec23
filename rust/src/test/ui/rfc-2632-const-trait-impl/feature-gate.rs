// revisions: stock gated
// gate-test-const_trait_impl

#![cfg_attr(gated, feature(const_trait_impl))]
#![allow(incomplete_features)]
#![feature(rustc_attrs)]

struct S;
trait T {}
impl const T for S {}
//[stock]~^ ERROR const trait impls are experimental

#[rustc_error]
fn main() {} //[gated]~ ERROR fatal error triggered by #[rustc_error]
