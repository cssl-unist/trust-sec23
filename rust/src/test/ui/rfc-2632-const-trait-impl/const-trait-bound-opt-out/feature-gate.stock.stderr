error[E0658]: `?const` on trait bounds is experimental
  --> $DIR/feature-gate.rs:13:29
   |
LL | const fn get_assoc_const<S: ?const T>() -> i32 { <S as T>::CONST }
   |                             ^^^^^^
   |
   = note: see issue #67794 <https://github.com/rust-lang/rust/issues/67794> for more information
   = help: add `#![feature(const_trait_bound_opt_out)]` to the crate attributes to enable

error: aborting due to previous error

For more information about this error, try `rustc --explain E0658`.
